src/internal/collection/postgres_collection.go

```diff
@@ -351,30 +430,72 @@ func (c *postgresCollection) list(
 		return nil
 	}
 
-	result := &model{}
-	for rows.Next() {
-		if err := rows.StructScan(result); err != nil {
-			return c.mapSQLError(err, "")
+	// To avoid holding the DB connection open for an unknown duration
+	// dictated by the client's callback, we:
+	// (1) buffer the SQL rows
+	// (2) close the connection
+	// (3) apply f(), the client's callback, to results in the buffer
+	// (4) if the buffer was full, re-execute the query, offset by key, and repeat (1)
+	bufferResults := func(rs *sqlx.Rows) ([]*model, int, error) {
+		defer rs.Close()
+		var rowCnt int
+		rowsBuffer := make([]*model, 0, c.listBufferCapacity)
+		result := &model{}
```

Comment:
You could just declare the model in the for loop since it is reset each iteration.